src/components/equipment/CreateEditEquipment.tsx

```diff
@@ -108,6 +113,21 @@ export default function CreateEditEquipment() {
     return <Loader container />;
   }
 
+  const hasEditAccess =
+    currentRole === UserRole.USER_OFFICER ||
+    isEquipmentResponsiblePerson(equipment, loggedInUser, currentRole) ||
+    isEquipmentOwner(equipment, loggedInUser);
+
+  if (id && !hasEditAccess) {
+    history.goBack();
+
+    return (
```

Comment:
if you call history.back(), maybe the return should be `null` as noone will see the view anyway, no?